esphome/components/mdns/__init__.py

```diff
@@ -79,6 +84,14 @@ async def to_code(config):
         elif CORE.is_rp2040:
             cg.add_library("LEAmDNS", None)
 
+    if CORE.using_esp_idf and (framework_ver.major, framework_ver.minor) >= (5, 0):
```

Comment:
Minor nit; but you can compare `cv.Version` instances directly:
```suggestion
    if CORE.using_esp_idf and framework_ver >= cv.Version(5, 0, 0):
```
(or maybe even drop the `framework_ver` variable).